source/MRMesh/MRAddNoise.cpp

```diff
@@ -0,0 +1,40 @@
+#include "MRAddNoise.h"
+
+namespace MR
+{
+
+void addNoise( VertCoords& points, const VertBitSet& validVerts, float sigma, unsigned int seed )
+{
+    if ( validVerts.count() > 100000 )
```

Comment:
I guess that speed up from multithreading can be reached with much smaller sizes, did you measure it?
Alternatively just keep only multithreaded branch, we do not care much about too small number of points.